### PR TITLE
Add pattern title to the Pattern Block sidebar panel heading

### PR DESCRIPTION
**Pull request by dreamwhisper**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/studiopress/pattern-manager/wiki/For-Developers#contributor-license-agreement-cla) with WP Engine.

### Summary of changes

Adds the pattern title to the Pattern Block sidebar panel.

### Related Issues
Fixes https://wpengine.atlassian.net/browse/GF-3805

### How to test

1. Add a Pattern Block in the pattern editor.
2. View the information presented in the sidebar. It should show `Pattern` when no pattern is selected yet. Once a pattern is selected, it will display the pattern title.

### Before:

Screenshot 2023-06-15 at 12 09 13 PM

### After:

Screenshot 2023-06-15 at 11 41 02 AM

### CODE REVIEW

**Comment by kienstra** [comment]
Good idea, reviewing this now.

File: wp-modules/editor/js/src/blocks/pattern-block/PatternEdit.tsx [hunk 1]
```diff
@@ -41,7 +41,11 @@ function PatternInspector( { pattern }: PatternInspectorProps ) {
 		<InspectorControls>
 			<Panel>
 				<PanelBody
-					title={ __( 'Pattern', 'pattern-manager' ) }
+					title={
+						pattern
+							? pattern.title
+							: __( 'Pattern', 'pattern-manager' )
```
Comment (kienstra) [inline wp-modules/editor/js/src/blocks/pattern-block/PatternEdit.tsx:47]: This isn't better, but here's another way to do this:

```js
title={ pattern?.title ?? __( 'Pattern', 'pattern-manager' ) }
```

Don't change this.

I just thought I'd share it, as I'm learning about the `??` operator.

Comment (dreamwhisper) [reply wp-modules/editor/js/src/blocks/pattern-block/PatternEdit.tsx:47]: Thanks @kienstra! 

Appreciate the info too!

Comment (kienstra) [reply wp-modules/editor/js/src/blocks/pattern-block/PatternEdit.tsx:47]: Thanks!

**Pull request merged**